common-content/en/module/complexity/worked-example-duplicate-encoder/index.md

```diff
@@ -0,0 +1,119 @@
++++
+title = "Worked example: Duplicate Encoder"
+time = 90
+emoji = "🧰"
+[build]
+  render = "never"
+  list = "local"
+  publishResources = false
+objectives = [
+    "Analyse the time complexity of a function.",
+    "Compare two algorithms for solving the same problem in terms of complexity.",
+    "Identify that speed isn't the only factor in choosing the best code.",
+]
++++
+
+Let's consider two possible approaches to [the Duplicate Encoder Codewars exercise](https://www.codewars.com/kata/54b42f9314d9229fd6000d9c/train/javascript).
+
+The problem can be stated as:
+
+{{<note type="exercise">}}
+The goal of this exercise is to convert a string to a new string where each character in the new string is `(` if that character appears only once in the original string, or `)` if that character appears more than once in the original string. Ignore capitalization when determining if a character is a duplicate.
```

Comment:
This sentence is insanely long
```suggestion
The goal of this exercise is to convert a string to a new string where, for each character:

1. if the original character appears only once, replace with `(``
1. if the original character appears more than once, replace with `)` 

Ignore capitalisation when deciding if a character is a duplicate.
```

Reply:
Yeah, this was me literally copy+pasting the text from Codewars - I've just rewritten to not reference Codewars :)